Added parsing of excel and semantics to header braces in tabledoc (#539)
EMMC-ASBL/tripper: docs/datadoc/documenting-a-resource.md

Reviewer: ```suggestion
corresponds to the following turtle representation:
```

The [TableDoc] class can be used to document multiple resources as rows in a table.

The table must have a header row with defined keywords (either [predefined][predefined keywords] or provided with a custom context).

> Tripper allow to have multiple columns with the same keyword in the header.

Reviewer: ```suggestion
Tripper allows multiple columns with the same keyword in the header.
```

Nested fields may be specified as dot-separated keywords.
Ex: `distribution.downloadURL`.

The header keyword(s) may be followed by an optional square bracket with the following syntax (inspired by URL options):

    name[label?var1=value1&var2=value2]

where
- `name`: is the column name that is mapped to a keyword (or set of dot-separated keywords) defined in the JSON-LD context.

>   The only formal requirement is that is cannot contain a begin brace ([), but it might be wise to be more strict.

Reviewer: ```suggestion
  The only formal requirement is that it cannot contain a begin brace ([), but it might be wise to be more strict.
```

- `label`: is a label for the column.
  It is used to make the column unique or to group related columns.

>   The only formal requirement is that is cannot contain a question mark (?), but it probably wise to be more strict.

Reviewer: ```suggestion
  The only formal requirement is that is cannot contain a question mark (?), but it is probably wise to be more strict.
```

  A digit should be allowed.
- `key`: a key identifying an option for the column.
  Should be a valid C or Python identifier (regex: `[_a-zA-Z][_a-zA-Z0-9]*`).
- `value`: the value of a key.
  Should not contain (unescaped) ampersand (&) or end braces (]).

Currently recognised keys:

> - **`unit`**: A unit symbol. All numbers in this column has this unit.

Reviewer: ```suggestion
- **`unit`**: A unit symbol. All numbers in this column have this unit.
```
